utilities/user_utils.py

```diff
@@ -155,96 +65,26 @@ def _create_htpasswd_file(username: str, password: str) -> tuple[Path, str]:
-            namespace="openshift-config",
-            htpasswd=htpasswd_b64,
-            type="Opaque",
-            wait_for_resource=True,
-        ) as secret:
-            yield secret
-    finally:
-        # Clean up the temporary file
-        temp_path.unlink(missing_ok=True)
-
-
-@contextmanager
-def _update_oauth_config(idp_name: str, secret_name: str) -> Generator[None, None, None]:
-    """
-    Updates the OpenShift OAuth configuration to add a new HTPasswd identity provider.
-    Uses ResourceEditor context manager to handle cleanup.
-
-    Args:
-        idp_name: The name of the identity provider to add
-        secret_name: The name of the secret containing the htpasswd file
-
-    Yields:
-        None, but ensures cleanup through ResourceEditor context manager
-    """
-    # Get current providers and add the new one
-    oauth = OAuth(name="cluster")
-    current_oauth = oauth.instance
-    identity_providers = current_oauth.spec.identityProviders
-
-    new_idp = {
-        "name": idp_name,
-        "mappingMethod": "claim",
-        "type": "HTPasswd",
-        "htpasswd": {"fileData": {"name": secret_name}},
-    }
-    updated_providers = identity_providers + [new_idp]
-
-    LOGGER.info("Updating OAuth")
-    with ResourceEditor(patches={oauth: {"spec": {"identityProviders": updated_providers}}}) as _:
-        # Wait for OAuth server to be ready
-        wait_for_pods_running(
-            admin_client=get_client(), namespace_name="openshift-authentication", number_of_consecutive_checks=1
-        )
-        LOGGER.info(f"Added IDP {idp_name} to OAuth configuration")
-        yield
-
-
-def _create_user_context(username: str, password: str, cluster_url: str) -> Optional[str]:
-    """
-    Attempts to login to OpenShift to create a user context.
+    Attempts to login to OpenShift
 
     Args:
         username: The username to login with
         password: The password to login with
         cluster_url: The OpenShift cluster URL
 
     Returns:
-        The user context name if login successful, None otherwise
-
-    Note:
-        This function is meant to be used with TimeoutSampler for retries.
-        It will return None on failure, allowing the sampler to retry.
+        True if login is successful
     """
     LOGGER.info(f"Attempting to login as {username}")
-
-    if login_with_user_password(api_address=cluster_url, user=username, password=password):
-        # Login was successful, get the current context
-        _, stdout, _ = run_command(command=["oc", "config", "current-context"], check=True)
-        user_context = stdout.strip()
-        LOGGER.info(f"Successfully created context for user {username}: {user_context}")
-        return user_context
-
-    LOGGER.error(f"Login failed for user {username}")
-    return None
+    res = login_with_user_password(api_address=cluster_url, user=username, password=password)
+    login_with_user_password(api_address=cluster_url, user=original_user)
```

Comment:
I don't understand why you log in again

Reply:
They should be together. User does not have same priviledges as admin user, so whereever we are logging in as the user, if that is successful, we should undo it. Do we want me to add a try/finally block?